qiskit/circuit/_utils.py

```diff
@@ -117,8 +119,9 @@ def with_gate_array(base_array):
     nonwritable = numpy.array(base_array, dtype=numpy.complex128)
     nonwritable.setflags(write=False)
 
-    def __array__(_self, dtype=None):
-        return numpy.asarray(nonwritable, dtype=dtype)
+    def __array__(_self, dtype=None, copy=_numpy_compat.COPY_ONLY_IF_NEEDED):
```

Comment:
On the previous `__array__`, the default of `copy` is `None`, here it is `_numpy_compat.COPY_ONLY_IF_NEEDED` is that fine?

Reply:
If we pass the `copy` argument directly to a Numpy function, we need to use the compatibility variable so that if a user has Numpy 1.x installed, they won't get errors (passing `copy=None` is an error in Numpy 1.x).  If we don't use the `copy` argument, or we only look at it ourselves, it's fine to use `None`.  Numpy 1.x never sets the `copy` argument to `__array__` (because it doesn't exist in the protocol in 1.x).

Reply:
Oh, I just realised that the previous `__array__` you're referring to is the one in `qiskit.circuit`. That's actually just an example in the middle of a docstring, so matching the now-suggested Numpy API exactly is probably best.